### PR TITLE
[DO NOT MERGE] Introducing Per-Site Waf Policy Support

### PR DESCRIPTION
**Pull request by venk8**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
`network-july-release` has been merged, please rebase and retarget to `master`

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json [hunk 1]
```diff
@@ -1513,6 +1513,10 @@
         "provisioningState": {
           "type": "string",
           "description": "Path rule of URL path map resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        },
+        "firewallPolicy": {
+          "$ref": "./network.json#/definitions/SubResource",
+          "description": "Reference of the FirewallPolicy resource."
```
Comment (argenet) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json]: (minor) "Reference TO..."?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json [hunk 2]
```diff
@@ -1681,6 +1685,10 @@
           "$ref": "./network.json#/definitions/SubResource",
           "description": "Redirect configuration resource of the application gateway."
         },
+        "firewallPolicy": {
```
Comment (argenet) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json]: For routing rules of 'Basic' type we should add a map of new objects.
That new object should be similar to ApplicationGatewayPathRulePropertiesFormat but only have 
`"paths"` and `"firewallPolicy"`. We can have multiple path prefixes with different WAF policies per a single Basic rule.

**Review by vkrivopalov — CHANGES_REQUESTED** [review]
We also need to add `firewallPolicy` to ApplicationGatewayHttpListenerPropertiesFormat for the per-site case. The current cases cover per-URI scenario (routing rules) only.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json (hunk 1 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json]: Done!

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json (hunk 2 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/applicationGateway.json]: Addressed!

**Comment by yungezz** [comment]
hi @venkatsvpr  @zikalino  what's the status on the PR?

**Pull request closed without merging**